go/enclave/components/interfaces.go

```diff
@@ -0,0 +1,139 @@
+package components
+
+import (
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+	gethrpc "github.com/ethereum/go-ethereum/rpc"
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/enclave/core"
+)
+
+type BlockIngestionType struct {
+	// IsLatest is true if this block was the canonical head of the L1 chain at the time it was submitted to enclave
+	// (if false then we are behind and catching up, expect to be fed another block immediately afterwards)
+	IsLatest bool
+
+	// Fork is true if the ingested block is on a different branch to previously known head
+	// (resulting in rewinding of one or more blocks that we had previously considered canonical)
+	Fork bool
+
+	// PreGenesis is true if there is no stored L1 head block.
+	// (L1 head is only stored when there is an L2 state to associate it with. Soon we will start consuming from the
+	// genesis block and then, we should only see one block ingested in a 'PreGenesis' state)
+	PreGenesis bool
+}
+
+type BlockConsumer interface {
+	ConsumeBlock(br *common.BlockAndReceipts, isLatest bool) (*BlockIngestionType, error)
+	GetHead() (*common.L1Block, error)
+}
+
+// Contains all of the data that each batch depends on
+type BatchContext struct {
+	BlockPtr     common.L1BlockHash // Block is needed for the cross chain messages
+	ParentPtr    common.L2BatchHash
+	Transactions common.L2Transactions
+	AtTime       uint64
+	Randomness   gethcommon.Hash
+	Creator      gethcommon.Address
+	ChainConfig  *params.ChainConfig
+}
+
+// ComputedBatch - a structure representing the result of a batch
+// computation where `Batch` is the newly computed batch and `Receipts`
+// are the receipts for the executed transactions inside this batch.
+// The `Commit` function allows for committing the stateDB resulting from
+// the computation of the batch. One might not want to commit in case the
+// resulting batch differs than what is being validated for example.
+type ComputedBatch struct {
+	Batch    *core.Batch
+	Receipts types.Receipts
+	Commit   func(bool) (gethcommon.Hash, error)
```

Comment:
We should include in one of the comments here what this bool parameter is. I only looked it up an hour ago and I've already forgotten :)